### PR TITLE
images/script: update the example hubble cli Deployment version

### PR DESCRIPTION
**Pull request by kaworu**
Update the Hubble CLI image version in the Deployment examples/ whenever we bump the Hubble CLI inside Cilium.

See https://github.com/cilium/cilium/pull/16459#discussion_r646756915 for context.

### CODE REVIEW

**Review by rolinh — APPROVED** [review]

File: images/scripts/update-hubble-version.sh [hunk 1]
```diff
@@ -34,10 +36,13 @@ for arch in amd64 arm64 ; do
...
+cat > "$hubble_version_script" << EOF
 # Code generated by images/scripts/update-hubble-version.sh; DO NOT EDIT.
 hubble_version="${hubble_version}"
 declare -A hubble_sha256
 hubble_sha256[amd64]="${hubble_sha256[0]}"
 hubble_sha256[arm64]="${hubble_sha256[1]}"
 EOF
+
+sed -e "s/\\(image: .*\\/hubble:\\).*/\1${hubble_version}/" "$hubble_cli_example" > "${hubble_cli_example}.tmp" &&
+    mv "${hubble_cli_example}.tmp" "$hubble_cli_example"
```
Comment (qmonnet) [inline images/scripts/update-hubble-version.sh]: Why not edit “in-place” with the `-i` option and avoid the `mv`? Are there portability issues?

Comment (rolinh) [reply images/scripts/update-hubble-version.sh]: @qmonnet I think it's more a question that if sed fails, the original file is left untouched and thus it lessens the chance of committing a broken file.

File: images/scripts/update-hubble-version.sh [hunk 2]
```diff
@@ -34,10 +36,13 @@ for arch in amd64 arm64 ; do
...
-cat > "${root_dir}/images/cilium/hubble-version.sh" << EOF
+cat > "$hubble_version_script" << EOF
 # Code generated by images/scripts/update-hubble-version.sh; DO NOT EDIT.
 hubble_version="${hubble_version}"
 declare -A hubble_sha256
 hubble_sha256[amd64]="${hubble_sha256[0]}"
 hubble_sha256[arm64]="${hubble_sha256[1]}"
 EOF
+
+sed -e "s/\\(image: .*\\/hubble:\\).*/\1${hubble_version}/" "$hubble_cli_example" > "${hubble_cli_example}.tmp" &&
```
Comment (gandro) [inline images/scripts/update-hubble-version.sh]: tiny nit (a personal preference to get rid of the `\\/`, feel free to ignore):

```suggestion
sed -e "s#\\(image: .*/hubble:\\).*#\1${hubble_version}#" "$hubble_cli_example" > "${hubble_cli_example}.tmp" &&
```

File: images/scripts/update-hubble-version.sh (hunk 1 above)
Comment (qmonnet) [reply images/scripts/update-hubble-version.sh]: What do you mean “if sed fails”? I don't expect it to ramdomly segfault, and even an I/O error in this context seems quite unlikely?

<details><summary>sed exit status</summary>

```
An exit status of zero indicates success, and a nonzero value indicates
failure.  GNU 'sed' returns the following exit status error values:

0
     Successful completion.

1
     Invalid command, invalid syntax, invalid regular expression or a
     GNU 'sed' extension command used with '--posix'.

2
     One or more of the input file specified on the command line could
     not be opened (e.g.  if a file is not found, or read permission is
     denied).  Processing continued with other files.

4
     An I/O error, or a serious processing error during runtime, GNU
     'sed' aborted immediately.
```
</details>

So if the expression is valid sed will apply it, else it doesn't and the file is untouched. The expression may not match what we had in mind, but that won't prevent sed to process the file entirely. Or am I missing something here?

(Asking out of curiosity, but honestly it matters very little and I'm also fine with the current version.)

**Review by gandro — APPROVED** [review]

File: images/scripts/update-hubble-version.sh (hunk 1 above)
Comment (kaworu) [reply images/scripts/update-hubble-version.sh]: > Why not edit “in-place” with the -i option and avoid the mv? Are there portability issues?

Correct, `-i` is not standard. Specifically, `-i suffix` works differently on OSX (suffix is not optional, one can use `sed -i '' …` for in-place but then we get ENOENT with GNU sed).

**Review by qmonnet — APPROVED** [review]

**Comment by kaworu** [comment]
Note to @cilium/janitors: no need to run CI on this PR.

File: images/scripts/update-hubble-version.sh (hunk 1 above)
Comment (qmonnet) [reply images/scripts/update-hubble-version.sh]: Ok thanks! For future reference, I found more info here https://unix.stackexchange.com/a/92907/70764.

**Pull request merged**